core/src/main/java/org/apache/calcite/sql2rel/RelFieldTrimmer.java

```diff
@@ -1336,6 +1338,144 @@ public TrimResult trimFields(
     return result(newSnapshot, inputMapping, snapshot);
   }
 
+  /**
+   * Trims {@link LogicalCorrelate} nodes.
+   */
+  public TrimResult trimFields(LogicalCorrelate correlate,
+      ImmutableBitSet fieldsUsed, Set<RelDataTypeField> extraFields) {
+    if (!extraFields.isEmpty() || correlate.getInputs().size() != 2) {
```

Comment:
Correlate is a BiRel, so the condition `correlate.getInputs().size() != 2` seems unnecessary.